### PR TITLE
Allow --create-log-groups when only some containers use awslogs

### PR DESCRIPTION
**Pull request by PettitWesley**
*Issue #, if available:*
#642

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by PettitWesley** [comment]
Testing:

```
version: '2'
services:
  container1:
    image: apache
    logging:
      driver: awslogs
      options:
        awslogs-group: somegroup
        awslogs-region: us-east-2
        awslogs-stream-prefix: somestream
  container2:
    image: apache
```

```
$ ecs-cli compose --file create-logs.yml service up --create-log-groups --cluster-config compose3
INFO[0000] Using ECS task definition                     TaskDefinition="compose3:12"
INFO[0001] Created Log Group somegroup in us-east-2     
INFO[0001] Updated the ECS service with a new task definition. Old containers will be stopped automatically, and replaced with new ones  desiredCount=1 force-deployment=false service=compose3
INFO[0001] Service status                                desiredCount=1 runningCount=1 serviceName=compose3
INFO[0064] (service compose3) has started 1 tasks: (task fa2e6bdd-1edd-4ae4-8d33-25fdf8821ada).  timestamp="2019-01-08 18:35:13 +0000 UTC"
```

File: ecs-cli/modules/cli/logs/logs_app.go [hunk 1]
```diff
@@ -309,6 +309,9 @@ func logConfigMisMatchError(taskDef *ecs.TaskDefinition, fieldName string) error
 // CreateLogGroups creates any needed log groups for the task definition to use CloudWatch Logs
 func CreateLogGroups(taskDef *ecs.TaskDefinition, logClientFactory cwlogsclient.LogClientFactory) error {
 	for _, container := range taskDef.ContainerDefinitions {
+		if container.LogConfiguration == nil || container.LogConfiguration.LogDriver == nil || aws.StringValue(container.LogConfiguration.LogDriver) != "awslogs" {
+			continue
+		}
 		logConfig, err := getContainerLogConfig(container)
 		if err != nil {
 			return err
```
Comment (allisaurus) [inline ecs-cli/modules/cli/logs/logs_app.go]: will this still fail if NO containers use awslogs? (and if so, can we add a test for it?)

Comment (PettitWesley) [reply ecs-cli/modules/cli/logs/logs_app.go]: It won't fail if no containers use awslogs- in that case nothing happens. Normally if log groups are created, then you'll get an Info message about it, if no containers use awslogs, then you get no message. I guess using `--create-log-groups` when no containers use awslogs is a user error- but I don't really want that to cause the command to fail. What do you think?

Comment (allisaurus) [reply ecs-cli/modules/cli/logs/logs_app.go]: I don't think it should fail either, though it feels like we may want to `WARN` in this case so customers know that option isn't doing anything. Would adding that check blow up the scope of the current change? If not let's add it, if so I don't consider it a blocker.

**Pull request merged**